numpy/ma/tests/test_arrayobject.py

```diff
@@ -27,14 +27,22 @@ def test_matrix_transpose_equals_transpose_2d():
 
 
 @pytest.mark.parametrize("shape", ARRAY_SHAPES_TO_TEST)
-def test_matrix_transpose_equals_swapaxes(shape):
+@pytest.mark.parametrize("kind", ["attribute", "func"])
+def test_matrix_transpose_equals_swapaxes(shape, kind):
     num_of_axes = len(shape)
     vec = np.arange(shape[-1])
     arr = np.broadcast_to(vec, shape)
 
     rng = np.random.default_rng(42)
     mask = rng.choice([0, 1], size=shape)
     ma_arr = masked_array(data=arr, mask=mask)
-
     tgt = np.swapaxes(arr, num_of_axes - 2, num_of_axes - 1)
```

Comment:
```suggestion
    tgt = np.swapaxes(ma_arr, num_of_axes - 2, num_of_axes - 1)
```

This is a bug in the test, I'll fix it tomorrow/Wednesday